mod/primitives/pkg/encoding/hex/bytes.go

```diff
@@ -29,17 +29,23 @@ import (
 var ErrInvalidHexStringLength = errors.New("invalid hex string length")
 
 // EncodeBytes creates a hex string with 0x prefix.
-// Inverse operation is ToBytes or MustToBytes.
+// Inverse operation is ToBytes or ToBytesSafe.
 func EncodeBytes(b []byte) string {
 	hexStr := make([]byte, len(b)*2+prefixLen)
 	copy(hexStr, Prefix)
 	hex.Encode(hexStr[prefixLen:], b)
 	return string(hexStr)
 }
 
-// MustToBytes returns the bytes represented by the given hex string.
-// It panics if the input is not a valid hex string.
-func MustToBytes(input string) []byte {
+// ToBytesSafe returns the bytes represented by the given hex string.
+// If the input is not a valid hex string, it returns an empty byte slice.
+func ToBytesSafe(input string) []byte {
```

Comment:
Currently we are only calling `hex.MustToBytes` in about dozen places in our codebase. I think in all places we can propagate any error with marshalling a string to bytes back to the caller to handle it, so we don't need to silently ignore any errors with the conversion.

So I would suggest  changing the function signature to something like:

```
func ToBytesSafe(input string) ([]byte, error)
```

Reply:
Yeah, Fair point! There's already a func  `ToBytes` doing this. Here I am referring it `https://github.com/berachain/beacon-kit/pull/2207#discussion_r1868763188 , we may end up using `ToBytes` everywhere.